ddtrace/appsec/_iast/_evidence_redaction/sql_sensitive_analyzer.py

```diff
@@ -0,0 +1,79 @@
+import re
+
+from ddtrace.appsec._iast.constants import DBAPI_MARIADB
+from ddtrace.appsec._iast.constants import DBAPI_MYSQL
+from ddtrace.appsec._iast.constants import DBAPI_PSYCOPG
+from ddtrace.appsec._iast.constants import DBAPI_SQLITE
+from ddtrace.internal.logger import get_logger
+
+
+log = get_logger(__name__)
+
+
+STRING_LITERAL = r"'(?:''|[^'])*'"
+POSTGRESQL_ESCAPED_LITERAL = r"\$([^$]*)\$.*?\$\1\$"
+MYSQL_STRING_LITERAL = r'"(?:\\\\"|[^"])*"|\'(?:\\\\\'|[^\'])*\''
+LINE_COMMENT = r"--.*$"
+BLOCK_COMMENT = r"/\*[\s\S]*?\*/"
+EXPONENT = r"(?:E[-+]?\\d+[fd]?)?"
+INTEGER_NUMBER = r"(?<!\w)\d+"
+DECIMAL_NUMBER = r"\d*\.\d+"
+HEX_NUMBER = r"x'[0-9a-f]+'|0x[0-9a-f]+"
+BIN_NUMBER = r"b'[0-9a-f]+'|0b[0-9a-f]+"
+NUMERIC_LITERAL = (
+    r"[-+]?(?:" + "|".join([HEX_NUMBER, BIN_NUMBER, DECIMAL_NUMBER + EXPONENT, INTEGER_NUMBER + EXPONENT]) + r")"
+)
+# ORACLE_ESCAPED_LITERAL = r"q'<.*?>'|q'\(.*?\)'|q'\{.*?\}'|q'\[.*?\]'|q'(?\<ESCAPE>.).*?\\k<ESCAPE>'"
+ORACLE_ESCAPED_LITERAL = r"q'<.*?>'|q'\(.*?\)'|q'\{.*?\}'|q'\[.*?\]'|q'([^']*)'"
+
+patterns = {
+    "ANSI": re.compile(
+        f"({NUMERIC_LITERAL})|({STRING_LITERAL})|({LINE_COMMENT})|({BLOCK_COMMENT})", re.IGNORECASE | re.MULTILINE
+    ),
+    DBAPI_MYSQL: re.compile(
+        f"({NUMERIC_LITERAL})|({MYSQL_STRING_LITERAL})|({LINE_COMMENT})|({BLOCK_COMMENT})", re.IGNORECASE | re.MULTILINE
+    ),
+    DBAPI_PSYCOPG: re.compile(
+        f"({NUMERIC_LITERAL})|({POSTGRESQL_ESCAPED_LITERAL})|({STRING_LITERAL})|({LINE_COMMENT})|({BLOCK_COMMENT})",
+        re.IGNORECASE | re.MULTILINE,
+    ),
+    "ORACLE": re.compile(
+        f"({NUMERIC_LITERAL})|({ORACLE_ESCAPED_LITERAL})|({STRING_LITERAL})|({LINE_COMMENT})|({BLOCK_COMMENT})",
+        re.IGNORECASE | re.MULTILINE,
+    ),
+}
+patterns[DBAPI_SQLITE] = patterns[DBAPI_MYSQL]
+patterns[DBAPI_MARIADB] = patterns[DBAPI_MYSQL]
+
+
+def sql_sensitive_analyzer(evidence, name_pattern, value_pattern):
```

Comment:
if it isn't already covered by one, a microbenchmark for this (and similar) functions could be useful. Especially if this is an area where you'd be interested in making performance improvements in the future.

Reply:
This refactor is covered by `flask_sqli` scenario :) :+1: